src/main/java/core/basesyntax/Record.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Record {
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private String originalData;
+    private LocalDate date;
+    private String userName;
+    private int countDays;
+    private int salary;
+
+    public Record(String data) {
+        this.originalData = data;
+        createFields();
+    }
+
+    private void createFields() {
+        if (originalData != null) {
```

Comment:
this check is redundant because if you get null here you will still create an object of class Record with all the null fields. So when you call recorder.getDate().isAfter() still NPE